packages/@uppy/aws-s3-multipart/src/index.js

```diff
@@ -86,13 +88,19 @@ class HTTPCommunicationQueue {
     }
     if (status === 403 && err.message === 'Request has expired') {
       if (!requests.isPaused) {
-        const next = this.#retryDelayIterator?.next()
-        if (next == null || next.done) {
-          return false
+        // We don't want to exhaust the retryDelayIterator as long as there are
+        // more than one request in parallel, to give slower connection a chance
+        // to catch up with the expiry set in Companion.
+        if (requests.limit === 1 || this.#previousRetryDelay == null) {
+          const next = this.#retryDelayIterator?.next()
+          if (next == null || next.done) {
+            return false
+          }
+          this.#previousRetryDelay = next.value
```

Comment:
so if I understand correctly, you only increase the retry interval if there is only one request left? Otherwise we have a very quick retry, all the time?

Reply:
We don't increase the retry interval, we take the next value in the iterator (which is user-provided, so we don't know if it increases, decreases, or stay the same).

If there are more than 1 request done in parallel, the RLQ limit is decreased and the failed request is requeued after waiting for a bit (again, the wait time is user provided, so we don't know if it's going to be "very quick" or not).
If there is only one request in parallel, the limit can't be decreased, so we iterate over `retryDelayIterator` as we do for other failures.

Should I add a comment explaining all that?

Reply:
yes I think a comment would help here. I was also a bit confused by the "delay" in `previousRetryDelay`, thinking it was the timeout on a glance

Reply:
Kit

Reply:
```suggestion
          // If there are more than 1 request done in parallel, the RLQ limit is
          // decreased and the failed request is requeued after waiting for a bit.
          // If there is only one request in parallel, the limit can't be
          // decreased, so we iterate over `retryDelayIterator` as we do for
          // other failures.
          // `#previousRetryDelay` caches the value so we can re-use it next time.
          this.#previousRetryDelay = next.value
```